### ISSUE TITLE
Best hyperparameters for all datasets

### ISSUE THREAD

**Issue by aarzchan** [facebookresearch/DPR#22]
Just as you did with Natural Questions, can you please also share the best hyperparameter settings for TriviaQA, WebQuestions, CuratedTREC, and SQuAD? This would help a lot in reproducing the results from the paper.

**Comment by shmsw25** [comment]
Hi @aarzchan, thanks for your interest!
For the DPR reader, we used the exactly same values of hyperparameters as NQ for TriviaQA and SQuAD. For WebQuestions and CuratedTREC, we used the batch size of 4 instead of 16, but other hyperparamters remain the same. We did minimal hyperparameter tuning, and the difference in batch size was mainly due to the difference in dataset size.

Passing it to @vlad-karpukhin for the retrieval hyperparameters. :)

**Comment by vlad-karpukhin** [comment]
Hi Aaron,
In addition to Sewon's comments, I can only confirm that we used same biencoder training parameters across all datasets except amount of train epochs.
We used 40 epochs for all 'large' datasets (NQ, Squad, Trivia) and 100 epochs for 'small' datasets (WebQuestions and CuratedTREC). In the combined biencoder training I upsampled small datasets 4 times.
The checkpoint selection was done using 'average rank' validation metric. 
Overall, biencoder training process is quite stable and you should get ~ same results in a wide range of hyperparameters values. We indeed didn't tune them much and there is a good chance you may get better results trying to tune those.

**Comment by aarzchan** [comment]
Thanks, @shmsw25 and @vlad-karpukhin! This helps a lot!

**Comment by niazarak** [comment]
@vlad-karpukhin Excuse me for replying in closed issue, I just don't want to create a separate one for a small comment.
> We used 40 epochs for all 'large' datasets (NQ, Squad, Trivia)
What about multi-dataset training?

**Comment by vlad-karpukhin** [comment]
Hi,

We used the same 40 epochs for the multiset setting,
but the small datasets (WebQ & Trec) were upsampled 4-5 times. And we
didn't use Squad for the multiset training.

Regards,
Vlad
